hummingbot/client/command/gateway_command.py

```diff
@@ -0,0 +1,194 @@
+        if option == "list-configs":
+            safe_ensure_future(self.show_gateway_connections())
+        elif option == "update":
+            safe_ensure_future(self.update_gateway(key, value))
+        elif option == "generate_certs":
+            safe_ensure_future(self._generate_certs())
+
+    async def _generate_certs(self,  # type: HummingbotApplication
+                              ):
+        if certs_files_exist():
+            self._notify(f"Gateway SSL certification files exist in {cert_path()}.")
+            self._notify("To create new certification files, please first manually delete those files.")
+            return
+        self.app.clear_input()
+        self.placeholder_mode = True
+        self.app.hide_input = True
+        while True:
+            pass_phase = await self.app.prompt(prompt='Enter pass phase to generate Gateway SSL certifications  >>> ',
+                                               is_password=True)
+            if pass_phase is not None and len(pass_phase) > 0:
+                break
+            self._notify("Error: Invalid pass phase")
+        create_self_sign_certs(pass_phase)
+        self._notify(f"Gateway SSL certification files are created in {cert_path()}.")
+        self.placeholder_mode = False
+        self.app.hide_input = False
+        self.app.change_prompt(prompt=">>> ")
+
+    async def generate_certs(self):
+        safe_ensure_future(self._generate_certs(), loop=self.ev_loop)
+
+    async def update_gateway(self, key, value):
+        safe_ensure_future(self._update_gateway(key, value), loop=self.ev_loop)
+
+    async def show_gateway_connections(self):
+        self._notify("\nTesting Gateway connections, please wait...")
+        safe_ensure_future(self._show_gateway_connections(), loop=self.ev_loop)
+
+    async def _api_request(self,
+                           method: str,
+                           path_url: str,
+                           params: Dict[str, Any] = {}) -> Dict[str, Any]:
+        """
+        Sends an aiohttp request and waits for a response.
+        :param method: The HTTP method, e.g. get or post
+        :param path_url: The path url or the API end point
+        :param params: A dictionary of required params for the end point
+        :returns A response in json format.
+        """
+        base_url = f"https://{global_config_map['gateway_api_host'].value}:" \
+                   f"{global_config_map['gateway_api_port'].value}"
+        url = f"{base_url}/{path_url}"
+        client = await self._http_client()
+        if method == "get":
+            if len(params) > 0:
+                response = await client.get(url, params=params)
+            else:
+                response = await client.get(url)
+        elif method == "post":
+            response = await client.post(url, data=params)
+
+        parsed_response = json.loads(await response.text())
+        return parsed_response
+
+    async def _http_client(self) -> aiohttp.ClientSession:
```

Comment:
I think we should have all Gateway API code outside of the command, this could be a Gateway base connector class (which later on will be inherited by other Gateway connectors). There should be `import aiohttp` in this gateway_command.py